### PR TITLE
[minor] Add DatatypeHelper func to split data by fold

### PR DESCRIPTION
**Pull request by moyapchen**
Use `tests/test_utils.py::TestDatatypeHelper::test_split_domains_by_fold` to check which tests a superset of this.

Doing so that I can make code for an internal class (one that uses data from a jsonl file) a little easier.

### CODE REVIEW

File: parlai/utils/data.py [hunk 1]
```diff
@@ -140,9 +135,30 @@ def split_domains_by_fold(
...
+        cls,
+        fold: str,
+        domains: List[List],
+        train_frac: float,
+        valid_frac: float,
+        test_frac: float,
+        seed: int = 42,
+    ):
+        """
+        Need to be careful about how we setup random to not leak examples between trains
```
Comment (stephenroller) [inline parlai/utils/data.py:152]: Please make a higher quality docstring

File: parlai/utils/data.py [hunk 2]
```diff
@@ -115,20 +115,15 @@ def is_streaming(cls, datatype: str) -> bool:
...
         fold: str,
-        domains: List[List],
+        data: List,
         train_frac: float,
         valid_frac: float,
         test_frac: float,
         seed: int = 42,
     ):
-        """
-        Need to be careful about how we setup random to not leak examples between trains
```
Comment (stephenroller) [inline parlai/utils/data.py:128]: Please make a  docstring for this method.

File: parlai/utils/data.py [hunk 3]
```diff
@@ -140,9 +135,30 @@ def split_domains_by_fold(
             start = train_frac + valid_frac
             end = 1.0
 
+        random.Random(seed).shuffle(data)
+        return data[int(start * len(data)) : int(end * len(data))]
+
+    @classmethod
+    def split_domains_by_fold(
```
Comment (stephenroller) [inline parlai/utils/data.py:142]: maybe shards? Or subsets? This is fine but `domains` is somewhat a specific name relative to how this function could be used?

**Review by stephenroller — APPROVED** [review]

File: parlai/utils/data.py (hunk 3 above)
Comment (moyapchen) [reply parlai/utils/data.py]: Changed to "subsets"

File: parlai/utils/data.py (hunk 1 above)
Comment (moyapchen) [reply parlai/utils/data.py]: Done.

File: parlai/utils/data.py (hunk 2 above)
Comment (moyapchen) [reply parlai/utils/data.py:128]: Done.

**Pull request merged**